docs/source/reference/udf.rst

```diff
@@ -29,6 +29,8 @@ Base UDF classes
     :members:
     :inherited-members:
 
+.. autoenum:: libertem.common.udf.UDFMethod
```

Comment:
I'm not sure what's the reason, but the `UDFMethod` enum is somehow not properly rendered in the return type position:

![image](https://github.com/LiberTEM/LiberTEM/assets/5778/178c3447-8998-42e1-9c3b-b6e65d6bea8b)

Just a note, and probably a bug in `enum-tools`?

Reply:
It looks like `Literal[Enum.member]` is generally problematic : https://github.com/sphinx-doc/sphinx/issues/11473

What I've done is create `UDFMethodT` alongside the enum, which is used to type annotate where necessary. This is slightly less useful in the doc but at least renders correctly, and there is always a link to the `UDFMethod` doc itself within the docstring.

Reply:
Ok, another question: why do we have `Literal[TheEnum.Member1, TheEnum.Member2, ...]` instead of just `TheEnum` as type? That should only be needed if we somehow only support a subset of variants of the `enum`, right? Or is there another reason?

Reply:
I was under the impression that that `def fn(...)-> MyEnum:` implies we return (an instance of) the Enum itself, not a member of the Enum (e.g. https://stackoverflow.com/a/67292548). I will see if I can find an official bit of doc...

Reply:
This is the documentation I was looking at: https://mypy.readthedocs.io/en/stable/literal_types.html#enums

This typechecks:

```python
from enum import Enum


class MyEnum(Enum):
    VariantOne = 1
    VariantTwo = 2


def f(my: MyEnum) -> MyEnum:
    return MyEnum.VariantTwo


f(MyEnum.VariantTwo)
```

I understood the Stackoverflow question as being about the actual string literals, like if we wanted to support strings like `"frame"` as return values

Reply:
Clearly I misunderstood, I've switched it to returning `UDFMethod` and removed the alias. Thanks!